components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/dao/SAMLArtifactDAO.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.sso.saml.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.joda.time.DateTime;
+import org.wso2.carbon.identity.core.persistence.JDBCPersistenceManager;
+import org.wso2.carbon.identity.sso.saml.dto.SAMLArtifactResolveDTO;
+import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOAuthnReqDTO;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+
+/**
+ * DAO class used to manipulate SAML artifacts data in the database.
+ */
+public class SAMLArtifactDAO {
+
+    private static final Log log = LogFactory.getLog(SAMLArtifactDAO.class);
+
+    private static final String ARTIFACT_STORE_SQL = "INSERT INTO IDN_SAML2_ARTIFACT_STORE(SOURCE_ID, " +
+            "MESSAGE_HANDLER, AUTHN_REQ_DTO, SESSION_ID, INIT_TIMESTAMP, EXP_TIMESTAMP) VALUES (?, ?, ?, ?, ?, ?)";
+    private static final String ASSERTION_RETRIEVE_SQL = "SELECT ID, AUTHN_REQ_DTO, SESSION_ID, INIT_TIMESTAMP, " +
+            "EXP_TIMESTAMP FROM IDN_SAML2_ARTIFACT_STORE WHERE SOURCE_ID=? AND MESSAGE_HANDLER=?";
+    private static final String ASSERTION_DELETE_SQL = "DELETE FROM IDN_SAML2_ARTIFACT_STORE WHERE ID=?";
+
+    private static final String ID_COLUMN_NAME = "ID";
+    private static final String AUTHN_REQ_DTO_COLUMN_NAME = "AUTHN_REQ_DTO";
+    private static final String SESSION_ID_COLUMN_NAME = "SESSION_ID";
+    private static final String INIT_TIMESTAMP_COLUMN_NAME = "INIT_TIMESTAMP";
+    private static final String EXP_TIMESTAMP_COLUMN_NAME = "EXP_TIMESTAMP";
+
+    /**
+     * Store SAML artifact in the database.
+     *
+     * @param artifactResolveDTO SAMLArtifactResolveDTO object with all data.
+     */
+    public void storeArtifact(SAMLArtifactResolveDTO artifactResolveDTO) {
+
+        try (Connection connection = JDBCPersistenceManager.getInstance().getDBConnection();
```

Comment:
Can we use JDBC template if possible for DAO layer. It will reduce lots of code lines.